server/controllers/routes/user/password.js

```diff
@@ -0,0 +1,36 @@
+const { compare } = require('bcryptjs');
+
+const { hashPassword } = require('../../utils/helper');
+const passwordSchema = require('./../../utils/passwordSchema');
+const {
+  updatePasswordQuery,
+} = require('./../../../database/queries/updatePassword');
+const { getPassword } = require('./../../../database/queries/getPassword');
+
+exports.updatePassword = (req, res) => {
+  const { oldPassword, newPassword } = req.body;
+  getPassword(req.user.id)
+    .then(dbRes => compare(oldPassword, dbRes.rows[0].password))
+    .then((passMatch) => {
+      if (passMatch) return passwordSchema.isValid({ password: newPassword });
+      return res.status(401).send({
```

Comment:
In case passwords don't match, this response will be sent, but still, the below code will be executed and error will be thrown
```suggestion
      const errObj = new Error('Old password is wrong')
      errObj.statusCode = 400,
      throw errObj
```

Reply:
Done